src/utils/getPartialCheckFromImportDiffs.ts

```diff
@@ -0,0 +1,89 @@
+import { reportWarning } from '../core/result';
+import NormalizedPath from '../types/NormalizedPath';
+import { FenceAndImportDiffs } from './getFenceAndImportDiffsFromGit';
+import * as path from 'path';
+import { PartialCheck } from '../types/PartialCheck';
+
+export function getPartialCheckFromImportDiffs(
+    graphDiff: FenceAndImportDiffs
+): PartialCheck | null {
+    let fences = new Set<NormalizedPath>();
+    let sourceFiles = new Set<NormalizedPath>();
+
+    let canResolve = true;
+
+    for (let [normalizedSourceFilePath, importDiff] of graphDiff.sourceImportDiffs.entries()) {
+        if (importDiff.addedImports?.length) {
+            // we need to re-check this file, since the new imports
+            // might violate the importing fence.

```

Comment:
Not just the importing fence(s), but also the exporting fence(s) around the imported module.  I believe this does happen correctly (right?), it's just the comment that's incomplete.

Reply:
Right, this comment assumes that the reader knows that re-checking this file means checking each of its imports against the imported module's export rules. I'll add a clarifying comment for new readers.